internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -280,48 +280,61 @@ func (a *Adapter) EnsureIntegrationPipelineRunsExist() (controller.OperationResu
 // EnsureGlobalCandidateImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the Snapshot is created
 func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResult, error) {
-	if a.component == nil || !gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
-		a.logger.Info("The Snapshot wasn't created for a single component push event, not updating the global candidate list.")
+	if (a.snapshot.GetLabels()[gitops.SnapshotTypeLabel] != gitops.SnapshotComponentType ||
+		!gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot)) && !gitops.IsOverrideSnapshot(a.snapshot) {
+		a.logger.Info("The Snapshot was neither created for a single component push event nor override type, not updating the global candidate list.")
 		return controller.ContinueProcessing()
 	}
+
+	if gitops.IsOverrideSnapshot(a.snapshot) && !gitops.HaveAppStudioTestsSucceeded(a.snapshot) {
+		a.logger.Info("This is a override snapshot but hasn't yet passed all required integration tests, not updating the global candidate list.")
+		return controller.ContinueProcessing()
+	}
+
 	if gitops.IsSnapshotMarkedAsAddedToGlobalCandidateList(a.snapshot) {
 		a.logger.Info("The Snapshot's component was previously added to the global candidate list, skipping adding it.")
 		return controller.ContinueProcessing()
 	}
 
-	for _, component := range a.snapshot.Spec.Components {
-		if component.Name == a.component.Name {
-			patch := client.MergeFrom(a.component.DeepCopy())
-			a.component.Spec.ContainerImage = component.ContainerImage
-			err := a.client.Patch(a.context, a.component, patch)
+	for _, snapshotComponent := range a.snapshot.Spec.Components {
+		component, err := a.loader.GetComponent(a.context, a.client, snapshotComponent.Name, a.snapshot.Namespace)
+		if err != nil {
+			_, loaderError := h.HandleLoaderError(a.logger, err, fmt.Sprintf("Failed to get component %s from applicaion, won't update global candidate list for this component", snapshotComponent.Name), a.application.Name)
+			if loaderError != nil {
+				return controller.RequeueWithError(loaderError)
+			}
+			continue
+		}
+
+		patch := client.MergeFrom(component.DeepCopy())
+		component.Spec.ContainerImage = snapshotComponent.ContainerImage
```

Comment:
It seems like this change updates all components for both the `component` and `override` Snapshots.

I think we should update the `spec.containerImage` only for the Snapshot's component in case of the `component` Snapshot.

Reply:
Thanks. Updated. I thought `override` snapshot will not have `.Status.LastBuiltCommit`. I add code `if !gitops.IsOverrideSnapshot(a.snapshot) {` to only update `containerImage`  for the component snapshot. Thanks.

Reply:
If I'm reading this right, the current code still updates `spec.containerImage` fields for all components, which I still think should happen only in case of `override` Snapshots.

Reply:
But in our old `component` snapshot GCL, we need to update the component's `spec.containerImage` [here](https://github.com/konflux-ci/integration-service/blob/main/internal/controller/snapshot/snapshot_adapter.go#L294-L309), I feel we should cover it, right?

Reply:
I think the latest changes work well for both cases, thanks!